### PR TITLE
warn when autoyast have wrong loader type

### PR DESCRIPTION
**Pull request by jreidinger**
trello: https://trello.com/c/hjG3dkgA/1431-2-bootloader-in-autoupgrade-autoinstallation-tell-the-user-that-their-profile-is-wrong-instead-of-throwing-an-exception-please

TODO:

- [x] changes
- [x] manual test

screenshot:
snimek obrazovky_2018-05-04_14-39-44

### CODE REVIEW

File: src/lib/bootloader/auto_client.rb [hunk 1]
```diff
@@ -27,7 +29,18 @@ def run
     end
 
     def import(data)
-      Yast::Bootloader.Import(data)
+      begin
+        Yast::Bootloader.Import(data)
+      rescue Bootloader::UnsupportedBootloader => e
+        possible_values = BootloaderFactory.supported_names + ["default"]
```
Comment (lslezak) [inline src/lib/bootloader/auto_client.rb]: Um, I do not like this magic string "default", it should be a constant...

File: src/lib/bootloader/auto_client.rb [hunk 2]
```diff
@@ -27,7 +29,18 @@ def run
...
+        AutoInstall.issues_list.add(:invalid_value, "bootloader", "loader_type",
+            e.bootloader_name,
+            _("Bootloader is not supported on this architecture. Possible values: ") +
```
Comment (lslezak) [inline src/lib/bootloader/auto_client.rb]: `Bootloader is not supported on this architecture` it a generic message saying that bootloaders do not work here. That's not true.

Proposals:

- `The selected bootloader is not supported on this architecture.`
- Or if you know what was in the profile you could include is as well:
  `The selected bootloader ('%{bootloader}') is not supported on this architecture.`

File: src/lib/bootloader/auto_client.rb [hunk 3]
```diff
@@ -27,7 +31,19 @@ def run
     end
 
     def import(data)
-      Yast::Bootloader.Import(data)
+      begin
+        Yast::Bootloader.Import(data)
+      rescue Bootloader::UnsupportedBootloader => e
+        textdomain "bootloader"
+        possible_values = BootloaderFactory.supported_names + ["default"]
```
Comment (lslezak) [inline src/lib/bootloader/auto_client.rb]: Um, I do not like this magic string `"default"`, it should be a constant...

File: src/lib/bootloader/auto_client.rb [hunk 4]
```diff
@@ -27,7 +31,19 @@ def run
...
+        AutoInstall.issues_list.add(:invalid_value, "bootloader", "loader_type",
+            e.bootloader_name,
+            _("Bootloader is not supported on this architecture. Possible values: ") +
```
Comment (lslezak) [inline src/lib/bootloader/auto_client.rb]: `Bootloader is not supported on this architecture` it a generic message saying that bootloaders do not work here. That's not true.

Proposals:

- `The selected bootloader is not supported on this architecture.`
- Or if you know what was in the profile you could include is as well:
  `The selected bootloader ('%{bootloader}') is not supported on this architecture.`

Comment (jreidinger) [reply src/lib/bootloader/auto_client.rb]: it is already passed as parameter above. So I do not want to be duplicite, so first one looks good.

File: src/lib/bootloader/auto_client.rb [hunk 5]
```diff
@@ -27,7 +31,19 @@ def run
...
     def import(data)
-      Yast::Bootloader.Import(data)
+      begin
+        Yast::Bootloader.Import(data)
+      rescue Bootloader::UnsupportedBootloader => e
+        textdomain "bootloader"
+        possible_values = BootloaderFactory.supported_names + [BootloaderFactory::DEFAULT_KEYWORD]
+        AutoInstall.issues_list.add(:invalid_value, "bootloader", "loader_type",
+            e.bootloader_name,
+            _("The selected bootloader is not supported on this architecture. Possible values: ") +
```
Comment (lslezak) [inline src/lib/bootloader/auto_client.rb]: I just noticed it's a single line string. How is that rendered, is it displayed in a RichText? Will that fit the 80x25 ncurses screen?

File: src/lib/bootloader/bootloader_factory.rb [hunk 6]
```diff
@@ -18,6 +18,8 @@ class BootloaderFactory
       "grub2-efi"
     ].freeze
 
+    DEFAULT_KEYWORD = "default".freeze
```
Comment (lslezak) [inline src/lib/bootloader/bootloader_factory.rb]: IMHO A comment would nice here, form the name it's hard to guess what that means...

File: src/lib/bootloader/auto_client.rb (hunk 5 above)
Comment (jreidinger) [reply src/lib/bootloader/auto_client.rb]: I have no clue, it is up to autoyast how it will show it.

Comment (schubi2) [reply src/lib/bootloader/auto_client.rb]: It should fit. There will be line breaks: https://trello.com/c/g5GxOY2c/1416-5-ay-implement-checking-for-firewall-part-of-the-profile

File: package/yast2-bootloader.changes [hunk 7]
```diff
@@ -1,3 +1,10 @@
+-------------------------------------------------------------------
+Fri May  4 12:43:44 UTC 2018 - jreidinger@suse.com
+
+- Inform user when bootloader is in supported in autoyast profile
```
Comment (lslezak) [inline package/yast2-bootloader.changes]: "Inform user when an unsupported bootloader is defined in the AutoYaST profile, display the valid values"

**Pull request merged**